src/frontends/tensorflow_common/src/op/bincount.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "common_op_table.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/unsqueeze.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_bincount_op(const NodeContext& node) {
+    default_op_checks(node, 3, {"Bincount"});
+    auto arr = node.get_input(0);
+    auto size = node.get_input(1);
+    auto weights = node.get_input(2);
+
+    std::vector<int32_t> size_scalar;
+    get_const_input(node, 1, &size_scalar);
+    TENSORFLOW_OP_VALIDATION(node, size_scalar.size() == 1, "size must be a scalar");
+    int32_t size_scalar_val = size_scalar[0];
+    TENSORFLOW_OP_VALIDATION(node, size_scalar_val > 0, "size must be non-negative.");
+
+    // weights: A Tensor. Must be one of the following types: int32, int64, float32, float64. is an int32, int64,
+    // float32, or float64 Tensor with the same shape as arr, or a length-0 Tensor, in which case it acts as all weights
+    // equal to 1.
+    auto weights_type = weights.get_element_type();
+    TENSORFLOW_OP_VALIDATION(node,
+                             weights_type == element::f32 || weights_type == element::f64 ||
+                                 weights_type == element::i32 || weights_type == element::i64,
+                             "Weights type is not supported: ",
+                             weights_type);
+    TENSORFLOW_OP_VALIDATION(node,
+                             arr.get_shape() == weights.get_shape() || shape_size(weights.get_shape()) == 0,
+                             "The shape of the weights must either match the first input or 0.");
+
+    if (shape_size(weights.get_shape()) == 0) {
+        // fill weighs with 1, the same shape with arr
+        weights = make_shared<v0::Constant>(weights_type, arr.get_shape(), std::vector<int>{1});
+    }
+
+    // implementation
+    auto start = make_shared<v0::Constant>(weights_type, Shape{}, std::vector<int>{0});
+    auto range_size = make_shared<v0::Constant>(weights_type, Shape{}, std::vector<int>{size_scalar_val});
+    auto step = make_shared<v0::Constant>(weights_type, Shape{}, std::vector<int>{1});
+    auto range = make_shared<v4::Range>(start, range_size, step, element::i32);
```

Comment:
for compasion with `arr` using `Equal`, range should be of the same type as `arr` i.e. int32. So please create `start`, `stop` of `i32` type. `size` is already of `i32` type

Reply:
using `weights_type` for start can lead to accuracy loss after casting from weights_type to i32.

Reply:
thanks, really make sense

will fix these.